_rules/element-marked-decorative-is-not-exposed-46ca7f.md

```diff
@@ -0,0 +1,150 @@
+name: Element marked as decorative is not exposed
+rule_type: atomic
+description: |
+  This rule checks that elements marked as decorative are not exposed in the accessibility tree, or with a presentational role.
+accessibility_requirements: # Remove whatever is not applicable
+input_aspects:
+  - DOM tree
+acknowledgments:
+  authors:
+    - Jean-Yves Moyen
+htmlHintIgnore:
+  # https://www.npmjs.com/package/htmlhint
+  # (used with `npm test` to ensure validity of code snippets)
+  - 'alt-require'
+---
+
+## Applicability
+
+The rule applies to any element which is [marked as decorative][].
+
+## Expectation
+
+Each target element either is not [included in the accessibility tree][] or has a [semantic role][] of `none` or `presentation`.
+
+## Assumptions
+
+_There are currently no assumptions_
+
+## Accessibility Support
+
+Implementation of the [Presentational Roles Conflict Resolution][] differs slightly from one user agent to the other. Hence, some elements might be exposed by one user agent and not by another, and consequently might create accessibility issues only for some users. In any cases, triggering the conflict is a sign of bad practice.
+
+## Background
+
+Elements are normally [marked as decorative][] to convey intention of the author that they are [pure decoration][decorative] and thus shouldn't be exposed by assistive technologies. On the other hand, elements that are [focusable][] are important to know for anybody and should be exposed to assistive technologies; and elements that are defining any [global ARIA attribute][] indicate intention to communicate something to the assistive technologies (through the `aria-*` attribute). When an element is both [marked as decorative][] and either [focusable][] or defining a [global ARIA attribute][], a conflict arise between these two intentions. The [conflict is resolved][presentational roles conflict resolution] by exposing the element.
+
+Whenever such a conflict occurs, this indicates at the very least mismatching intentions. Such a conflict should be avoided.
+
+When these conflicts arise on [decorative][] [non-text content][], this is also a failure of [Success Criterion 1.1.1: Non-text Content][sc111] because [decorative][] [non-text content][] must be implemented in a way that allows assistive technologies to ignore it. When this conflicts arise on text content, or on content which is not [decorative][], this is not a failure of WCAG. Therefore this rule is not mapping to any specific WCAG Success Criterion, and is not an accessibility requirement for WCAG.
+
+- [Success Criterion 1.1.1: Non-text Content][sc111]
+- [Technique F38: Failure of Success Criterion 1.1.1 due to not marking up decorative images in HTML in a way that allows assistive technology to ignore them][f38]
+- [Technique F39: Failure of Success Criterion 1.1.1 due to providing a text alternative that is not null (e.g., alt="spacer" or alt="image") for images that should be ignored by assistive technology][f39]
```

Comment:
I don't think this should be here. I disagree with the assessment that it is a WCAG failure for images, as we discussed before. It might be, depending on the purpose of the image. There are other rules which would catch such issues. I think we are better off staying clear of any reference to WCAG in this rule, to avoid confusing this with actual WCAG issues.

```suggestion
```

Reply:
> I disagree with the assessment that it is a WCAG failure for images, as we discussed before. It might be, depending on the purpose of the image.

Yes. "**if** (some condition), this is a failure". We agreed that it was too big a if to be in an assumption. I feel it is still worth being mentioned. It's in the background, and it does explain a bit why the rule is here.

> There are other rules which would catch such issues.

I do not think we have anything that catches "decorative image not being implemented in a way that allows AT to ignore it".
Which other rule do you think about that would flag `<img src="decorative.jpg" role="none" aria-label="this is a totally decorative image" />`?

> I think we are better off staying clear of any reference to WCAG in this rule, to avoid confusing this with actual WCAG issues.

I think this does explain why the rule is here. We're not chasing after corner cases in specs to make a rule. This is something that is not **always** a problem (so, no accessibility mapping), but is nonetheless **sometimes** a problem (and we have no data on how often it is a problem). I feel keeping this explain why the rule is here.